lib/mobius/models/utils.ex

```diff
@@ -19,4 +19,19 @@ defmodule Mobius.Models.Utils do
 
     struct!(struct, [{struct_key, value}])
   end
+
+  @spec parse_list(any, (any -> output)) :: [output] | nil when output: var
```

Comment:
```suggestion
  @spec parse_list(input, (Enum.t(input) -> output)) :: [output] | nil when output: var, input: var
```

Not quite sure this is right, but I'm pretty sure we could a constraint for the input.

Reply:
We could if the function raised an error when the input isn't a list, but this function is perfectly valid for non-lists, it just returns nil

Also the input of the function isn't `Enum.t`, it's the individual elements of the list. The success typing of this function is actually `@spec parse_list([input], (input -> output)) :: [output] when input: var, output: var` which is what I had there until I added the nil return

Reply:
You're right, forgot about the nil return.